### ISSUE TITLE
Double input on form collection with sonata_admin_collection child admin

### ISSUE THREAD

**Issue by Soullivaneuh** [sonata-project/SonataAdminBundle#2909]
If I have a child admin class to be used by `sonata_admin_collection` type with some `collection` text fields like this:

```php
class SubClassAdmin extends Admin
{
    protected function configureRoutes(RouteCollection $collection)
    {
        $collection
            ->remove('list')
            ->remove('batch')
            ->remove('edit')
            ->remove('delete')
            ->remove('show')
            ->remove('export')
        ;
    }

    protected function configureFormFields(FormMapper $formMapper)
    {
        $formMapper
            ->add('emails', 'collection', [
                'type'          => 'text',
                'allow_add'     => true,
                'allow_delete'  => true,
            ])
        ;
    }

    public function showIn($context)
    {
        if ($context === Admin::CONTEXT_DASHBOARD) {
            return false;
        }

        return parent::showIn($context);
    }

}
```

And the parent class using `sonata_admin_collection` like this:

```php
class ClassAdmin extends Admin
{
    protected function configureFormFields(FormMapper $formMapper)
    {
        $context = $this->getPersistentParameter('context');

        $formMapper
                ->add('subClasses', 'sonata_type_collection', [
                    'by_reference'          => false,
                    'mapped'                => true,
                ], [
                    'edit'                  => 'inline',
                    'inline'                => 'table',
                    'link_parameters'       => ['context' => $context],
                ])
        ;
    }
}
```

This works an show child forms if needed. But if I want to add an email on this child form by clicking on the "Add" button, I got twice text inputs:

![sonata_child_collection_twice](https://cloud.githubusercontent.com/assets/1698357/7155498/765bd8f4-e35f-11e4-8f43-b2fb740b4861.png)

This bug appears only if you add new child form.

Probably a JS issue, I'll try to investigate later.

My sonata composer info:

```bash
$ composer info -i | grep sonata-project
sonata-project/admin-bundle              dev-master ad15a5b  Symfony SonataAdminBundle
sonata-project/block-bundle              2.3.0               Symfony SonataBlockBundle
sonata-project/cache                     1.0.5               Cache library
sonata-project/core-bundle               2.3.1               Symfony SonataCoreBundle
sonata-project/doctrine-extensions       1.0.2               Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle dev-nexylan 280ab07 Symfony Sonata / Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/easy-extends-bundle       2.1.9               Symfony SonataEasyExtendsBundle
sonata-project/exporter                  1.3.4               Lightweight Exporter library
sonata-project/intl-bundle               2.2.1               Symfony SonataIntlBundle
```

**Comment by malavsky** [comment]
Look at the Admin.js in SonataAdminBundle.
```
jQuery(document).ready(function() {
    jQuery('html').removeClass('no-js');
    if (window.SONATA_CONFIG && window.SONATA_CONFIG.CONFIRM_EXIT) {
        jQuery('.sonata-ba-form form').each(function () { jQuery(this).confirmExit(); });
    }

    Admin.setup_per_page_switcher(document);

    Admin.shared_setup(document);
});
```
```
Admin.shared_setup(document);
```
function `shared_setup`  call `setup_collection_buttons` function, which bind onClick event on `.sonata-collection-add` (plus button).

`sonata_type_collection` include file [edit_orm_one_association_script.html](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/master/Resources/views/CRUD/edit_orm_one_association_script.html.twig#L51).
And on success also call `Admin.shared_setup()` which again bind event for `.sonata-collection-add` (plus button)

**Comment by VadMat** [comment]
Hi everybody, is this solved? I'm having this issue with sonata_type_collection.

I have Sonata 3.20.1 and I still getting multiple inputs when I click add button. (Is strange, but always is the childs entity quantity  +1)

**Comment by jlamur** [comment]
This issue should be solved by #2892. If you still have the same bug, could you please open another issue with relevant code?
